CONTRIBUTING.md

```diff
@@ -1,178 +1,494 @@
 # Contributing
 Contributing encompasses repository specific requirements, and the global [Quipucords contribution guidelines](https://github.com/quipucords/quipucords/blob/master/CONTRIBUTING.md).
 
-## Install
-Before developing you'll need to install:
-* [NodeJS and NPM](https://nodejs.org/)
-* [Docker](https://docs.docker.com/desktop/)
-  * Alternatively, you can try [Podman](https://github.com/containers/podman).
-* And [Yarn](https://yarnpkg.com)
+## Process
+<details>
+<summary><h3 style="display: inline-block">Using Git</h3></summary>
+
+Quipucords-UI makes use of
+- GitHub's fork and pull workflow.
+- A linear commit process and rebasing. GitHub merge commits, and squashing are discouraged in favor of smaller independent commits
+
+> Working directly on the main repository is discouraged.
+
+#### Branch syncing
+Linear commit history for Quipucords-UI makes syncing concise
+- `dev` is always rebased from `main`
+  - typically after a release
+  - or in prep for a fast-forward of `main`
+- `main` is fast-forwarded from `dev`
+  - typically when commits are prepared for release
+
+</details>
+
+<details>
+<summary><h3 style="display: inline-block">Pull request workflow, and testing</h3></summary>
+
+All development work should be handled through GitHub's fork and pull workflow.
+
+#### Setting up a pull request
+When multiple developers are contributing features development pull requests (PRs) should be opened against the `dev` branch. PRs directly to `main` are discouraged, 
```

Comment:
suggestion

`When multiple developers are contributing features, development pull requests (PRs) should be opened against the `dev`branch.  `